app/services/CategorisationService.scala

```diff
@@ -39,122 +39,87 @@ class CategorisationService @Inject() (
+        if (catInfo.commodityCode == getGoodsRecordResponse.comcode || recategorising) {
+          Future.successful(request.userAnswers)
+        } else {
+          updateCategorisationDetails(
+            request,
+            recordId,
+            recordCategorisations,
+            getGoodsRecordResponse.comcode,
+            getGoodsRecordResponse.countryOfOrigin
+          )
+        }
+
+      case None =>
+        updateCategorisationDetails(
+          request,
+          recordId,
+          recordCategorisations,
+          getGoodsRecordResponse.comcode,
+          getGoodsRecordResponse.countryOfOrigin
+        )
+    }).flatten
 
-  def updateCategorisationWithNewCommodityCode(
+  private def updateCategorisationDetails(
     request: DataRequest[_],
-    recordId: String
-  )(implicit
-    hc: HeaderCarrier
-  ): Future[UserAnswers] = {
-
-    val recordCategorisations =
-      request.userAnswers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
-
-    val originalCommodityCodeOpt =
-      recordCategorisations.records.get(recordId).flatMap(_.originalCommodityCode)
-
+    recordId: String,
+    recordCategorisations: RecordCategorisations,
+    commodityCode: String,
+    countryOfOrigin: String
+  )(implicit hc: HeaderCarrier) =
     for {
-      newCommodityCode       <- Future.fromTry(Try(request.userAnswers.get(LongerCommodityQuery(recordId)).get))
-      getGoodsRecordResponse <- goodsRecordsConnector.getRecord(eori = request.eori, recordId = recordId)
-      goodsNomenclature      <- ottConnector.getCategorisationInfo(
-                                  newCommodityCode.commodityCode,
-                                  request.eori,
-                                  request.affinityGroup,
-                                  Some(recordId),
-                                  getGoodsRecordResponse.countryOfOrigin,
-                                  LocalDate.now() //TODO where does DateOfTrade come from??
-                                )
-      originalCommodityCode   = originalCommodityCodeOpt.getOrElse(getGoodsRecordResponse.comcode)
-      categorisationInfo     <- CategorisationInfo.build(goodsNomenclature, Some(originalCommodityCode)) match {
-                                  case Some(categorisationInfo) => Future.successful(categorisationInfo)
-                                  case _                        => Future.failed(new RuntimeException("Could not build categorisation info"))
-                                }
-      updatedAnswers         <-
+      goodsNomenclature       <- ottConnector.getCategorisationInfo(
+                                   commodityCode,
+                                   request.eori,
+                                   request.affinityGroup,
+                                   Some(recordId),
+                                   countryOfOrigin,
+                                   LocalDate.now() //TODO where does DateOfTrade come from??
```

Comment:
current date time makes sense here I believe, commodity data can change and I believe this date time is to get the data as of a certain date - in our case we want the latest data - can remove this comment if that makes sense to you

Reply:
Removed 🙂